### ISSUE TITLE
STORY: Upgrade to Elasticsearch 5.x

### ISSUE THREAD

**Issue by nickstenning** [hypothesis/product-backlog#271]
### Context

Our search index currently uses a relatively old version of Elasticsearch, v1.5.x.

It would make sense to upgrade to Elasticsearch v5.x for several reasons:

- to limit our exposure to deprecation of v1.5.x, especially given our use of a managed Elasticsearch service in Amazon
- Elasticsearch v2.x and v5.x both have not-insignificant performance improvements relative to v1.5.x
- friends of ours who are running their own `h` instance are currently having to run two Elasticsearch clusters side-by-side to support both `h` and other software they use: they'd like to converge on v5.x

### Task

Update `h` to run on Elasticsearch v5.x, and upgrade to a v5.x cluster in production.

### Discussion

It's possible we'll need to upgrade in two steps (from v1.5.x to v2.x, and then from v2.x from v5.x). If we can convince ourselves that we can safely make the jump in one step, this is probably preferable.

Elasticsearch provides an [`elasticsearch-migration` plugin](https://github.com/elastic/elasticsearch-migration) which we can use to help us identify issues with our indices which need addressing before we can upgrade. We'll likely need to use this in development as support for the migration plugin in AWS Elasticsearch Service seems to be inconsistent.

If we need to, we can probably accept some short downtime (30m) if it makes this move significantly easier in production. If we can find a straightforward way of migrating without downtime, that would be preferable.

**Comment by segdeha** [comment]
Closing this as it's superseded by https://github.com/hypothesis/product-backlog/issues/562 which is nearing completion.
